node/src/components/api_server/rpcs/state.rs

```diff
@@ -212,3 +222,138 @@ impl RpcWithParamsExt for GetBalance {
+                        return Ok(response_builder.error(warp_json_rpc::Error::custom(
+                            ErrorCode::NoSuchBlock as i64,
+                            error_msg,
+                        ))?);
+                    }
+                    Some(block) => block,
+                }
+            };
+
+            let protocol_version = ProtocolVersion::V1_0_0;
+            let protocol_version_result = effect_builder
+                .make_request(
+                    |responder| ApiRequest::QueryProtocolData {
+                        protocol_version,
+                        responder,
+                    },
+                    QueueKind::Api,
+                )
+                .await;
+
+            let protocol_data = {
+                if let Ok(Some(protocol_data)) = protocol_version_result {
+                    protocol_data
+                } else {
+                    Box::new(ProtocolData::default())
+                }
+            };
+
+            // auction contract key
+            let base_key = protocol_data.auction().into();
+            // bids named key in auction contract
+            let path = vec![casper_types::auction::BIDS_KEY.to_string()];
+            // the global state hash of the last block
+            let global_state_hash = *block.post_state_hash();
+            // the era of the last block
+            let era_id = block.header().era_id().0;
+
+            let query_result = effect_builder
+                .make_request(
+                    |responder| ApiRequest::QueryGlobalState {
+                        global_state_hash,
+                        base_key,
+                        path,
+                        responder,
+                    },
+                    QueueKind::Api,
+                )
+                .await;
+
+            let bids = {
+                if let Ok(QueryResult::Success(stored_value::StoredValue::CLValue(cl_value))) =
+                    query_result
+                {
+                    cl_value.into_t().ok()
+                } else {
+                    None
+                }
+            };
+
+            let era_validators_result = effect_builder
+                .make_request(
+                    |responder| ApiRequest::QueryEraValidators {
+                        global_state_hash,
+                        era_id,
+                        protocol_version,
+                        responder,
+                    },
+                    QueueKind::Api,
+                )
+                .await;
+
+            let validator_weights = {
+                if let Ok(validator_weights) = era_validators_result {
+                    validator_weights
+                } else {
+                    None
+                }
+            };
+
+            let auction_state =
+                AuctionState::new(global_state_hash, era_id, bids, validator_weights);
+            info!("AuctionState --auction_state: {:?}", auction_state);
```

Comment:
Was this left in by mistake?

Reply:
Never mind - I added that comment before your second commit.  I refreshed and my comment seemed to have disappeared, but apparently not :man_shrugging: